tests/operators/test_python.py

```diff
@@ -311,6 +312,314 @@ def func(**context):
         python_operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
 
 
+class TestAirflowTask(TestPythonBase):
+
+    def setUp(self):
+        super(TestAirflowTask, self).setUp()
+
+        # Methods need self and we don't support that.
+        @task_decorator
+        def do_run():
+            self.run = True
+
+        self.do_run = do_run
+
+    def is_run(self):
+        return self.run
+
+    def test_task_run(self):
+        """Tests that the python callable is invoked on task run."""
+
+        with self.dag:
+            self.do_run()
+        self.assertFalse(self.is_run())
```

Comment:
Can we try to avoid creating side effects on class level?

Reply:
Also, Airflow community has not yet migrated (planned after 2.0) to pytest way fo writing tests but it would be nice to get rid of unittests in new tests